rllib/evaluation/rollout_worker.py

```diff
@@ -1179,9 +1179,10 @@ def get_policy(self, policy_id: PolicyID = DEFAULT_POLICY_ID) -> Optional[Policy
     @DeveloperAPI
     def add_policy(
         self,
-        *,
         policy_id: PolicyID,
-        policy_cls: Type[Policy],
+        policy_cls: Optional[Type[Policy]] = None,
+        policy: Optional[Policy] = None,
```

Comment:
just want to mention a debate I had with myself while looking at this.
if we are willing to sacrifice a little bit of efficiency for the local worker, we can actually make PolicySpec the narrow waist of all this.
then we won't need to change rollout_worker or policy_map.
and in Algorithm.add_policy(), we would simple get the policy_spec and policy_state if we get passed a policy, instead of policy_cls.
I understand that will cause the policy to be created again for the local worker, wasting compute and mem. but it seems like we can greatly simplify all these logics if we can have a narrow waist for add_policy().
just some thoughts that I figure I'd mention, since I can't make up my mind either.
let me know what you think.

Reply:
Hmm, I'm not sure about this. Was thinking about this, too :)
But the problem is that the expectation (mental model of user) of doing `my_algo.add_policy(my_policy_instance)` is that `my_policy_instance` is actually as-is being incorporated somehow to the algorithm.

Reply:
yeah maybe.
if we were writing C++, we can make this super clear by declare the argument ``const Policy& policy`` if the policy is meant to be duplicated, or "Policy* policy" if the ownership of the policy is supposed to be transferred.
python is 🤷‍♂️

Reply:
🤷‍♂️

Reply:
:)